File: Laravel/app/Http/Requests/Api/Auth/RestoreFromMailRequest.php
<?php

namespace App\Http\Requests\Api\Auth;

use App\Http\Requests\ApiRequest;

class RestoreFromMailRequest extends ApiRequest
{
    public function rules(): array
    {
        return [
            'reset_token' => 'required|string',
            'email' => 'required|string|email|exists:users,email',
            'new_password' => 'required|string|min:6|confirmed',
            'new_password_confirmation' => 'required|string|min:6',
            'logout' => 'boolean',
        ];
    }
}

File: PSB/Api/Request/Mail/RestoreFromMailRequest.cs
using System.Text.Json.Serialization;

namespace PSB.Api.Request.Mail
{
    public class RestoreFromMailRequest(string email, string resetToken, string password, string passwordConfirmation, bool logout)
    {
        [JsonPropertyName("email")] public string Email { get; set; } = email;
        [JsonPropertyName("reset_token")] public string ResetToken { get; set; } = resetToken;
        [JsonPropertyName("new_password")] public string NewPassword { get; set; } = password;
        [JsonPropertyName("new_password_confirmation")] public string NewPasswordConfirmation { get; set; } = passwordConfirmation;
        [JsonPropertyName("logout")] public bool Logout { get; set; } = logout;

    }
}
